libs/storage/src/ipfs/ipfs.service.ts

```diff
@@ -59,32 +59,34 @@ export class IpfsService {
     return Buffer.concat(chunks);
   }
 
-  /**
-   * gets the true content-length of the CID file from the IPFS gateway,
-   * or NaN if the content-length is unknown.
-   * if the file does not exist locally, or a network error occurs, throws an error
-   */
-  public async contentLengthInLocalGateway(cid: string): Promise<number> {
+  public async getInfoFromLocalNode(cid: string): Promise<FilesStatResult> {
```

Comment:
(Took out the `checkForExistence` flag here, because this will now error (not hang) if the file doesn't exist in the local node, so we can save the extra network call)

As noted in the comment, the Kubo CLI maps 1:1 to the Kubo RPC API, with all supported CLI options mapped to properties... but for some reason, the JS package doesn't include a specific property for the `offline` option. We can explicitly override typing here as we do, or we could use the `timeout: number | string` property (milliseconds) to achieve the desired result.